native/cocos/platform/filesystem/IFileSystem.h

```diff
@@ -0,0 +1,89 @@
+
+/****************************************************************************
+ Copyright (c) 2022 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+
+#pragma once
+
+#include "base/Macros.h"
+#include "base/std/container/unordered_map.h"
+
+#include "cocos/platform/filesystem/FilePath.h"
+#include "cocos/platform/filesystem/IFileHandle.h"
+
+namespace cc {
+class CC_DLL IFileSystem {
+public:
+    enum class AccessFlag {
+        READ_ONLY  = 0x0000,
+        WRITE_ONLY = 0x0001 << 0,
+        READ_WRITE = 0x0001 << 1,
+        APPEND     = 0x0001 << 2,
+    };
```

Comment:
The flag takes effect for file, but define here is weird.

Reply:
Define it in IFileHandle is more suitable.

Reply:
The behavior of opening a file is in the FileSystem, and it seems that it is not a good idea to put it in the IFileHandle. IFileHandle is only the operation of the file content, the behavior of opening the file does not occur in it

Maybe put it globally?

Reply:
It is a state of a FileHandle object. So i think it is reasonable to put it in IFileHandle. Put it in global seems ok. And how about put all file system related codes into `fs` namespace?

Reply:
If it is opened in FileHandle, I think it is ok, but since the open operation will depend on FileSystem, I put it in FileSystem.

On the other hand, file opening, file moving, file deletion, and file naming are all operations on files. File read, file write, this is the operation of the file content